cfme/tests/intelligence/test_reports_with_timelines.py

```diff
@@ -1,39 +1,121 @@
-        assignee: jdupuy
-        initialEstimate: 1/5h
-        casecomponent: web_ui
-        caseimportance: medium
-        testSteps:
-            1. Add an infrastructure provider to ensure that an event for each report will occur
-            2. Navigate to Cloud Intel -> Reports
-            3. Modify one of the default reports to display events from today
-            4. Navigate to Cloud Intel -> Timelines
-            5. Click on the report you created
-            6. Verify that the timeline (with events) is properly displayed
+UPDATES = [
+    {
+        'filter': {'primary_filter': "fill_field(Policy Event : Activity Sample - Timestamp ("
+                                     "Day/Time), ""IS, This Week)"},
+    },
+    {'report_fields': ["Activity Sample - Timestamp (Day/Time)",
+                       "Miq Policy Sets : Date ""Created", "Miq Policy Sets : Date Updated",
+                       "Miq Policy Sets : Description", "Miq Policy Sets : EVM Unique ID (Guid)"],
+     'filter':
+        {'primary_filter': "fill_field(Policy Event : Activity Sample - Timestamp (Day/Time), "
+            "IS, This Week)"},
+     },
+    {'filter':
+        {'primary_filter': "fill_field(Host / Node : Date Created, IS, This Week)"}},
+    {'filter':
+        {'primary_filter': "fill_field(Event Stream.Host / Node : Total VMs, >, 0)"}},
+]
+
+
+@pytest.fixture(scope="function")
+def setup_for_reports():
+    """ wrapper function for _setup_for_reports, so that it runs during test
+
+    Returns:
+        unbound function object for calling during the test
     """
-    pass
+    def _setup_for_reports(request, appliance, provider, path, updates, vm_crud, register_event): # noqa
+        """ This function makes copies of the default reports so that we can test the timelines
+        on the CloudIntel->Timelines page
+
+        Args:
+            request: py.test funcarg request
+            appliance: IPAppliance object
+            provider: provider object
+            path: path to the report in the tree
+            updates: updates to the default report so that it will display events
+            vm_crud: vm for Policy events
+            register_event: function for registering events
+            args: kwargs for extra_steps
+        """
+        # first perform extra steps for Policy stuff
+        if "Policy" in path:
+            generate_policy_event(request, appliance, provider, vm_crud, register_event) # noqa
+        report = appliance.collections.reports.instantiate(
+            type=path[0], subtype=path[1], menu_name=path[2]
+        )
+        # copy the default report
+        copied_report = report.copy()
+        # update the copied report
+        copied_report.update(updates)
+        # delete report at end of test
+        request.addfinalizer(copied_report.delete)
+
+        return copied_report
+
+    return _setup_for_reports
+
+
+def generate_policy_event(request, appliance, provider, vm_crud, register_event): # noqa
```

Comment:
Question:
why this function? I found use of this in `L65` as per my observation... Is it possible to covert its as fixture for test?

Reply:
@digitronik This is a function because it is time consuming and I only want it to run if we are running the tests that depend on having a policy event having occurred. If it is a fixture, it will run at the start of each test, and it is unnecessary to run it for half the tests here.